src/pfe/file-watcher/server/test/functional-test/suites/project/tests/project-specification.ts

```diff
@@ -42,37 +43,81 @@ export function projectSpecificationTest(socket: SocketIO, projData: ProjectCrea
                 "setting": "internalDebugPort",
                 "socketEvent": eventConfigs.events.settingsChanged,
-                "eventKeys": ["operationId", "projectID", "status", "ports"]
+                "eventKeys": ["operationId", "projectID", "status", "ports"],
+                "result": {
+                    "projectID": projData.projectID,
+                    "status": "success",
+                    "ports": {
+                        "internalDebugPort": project_configs.exposedDebugPorts[0]
+                    }
+                },
             },
             "combo3": {
                 "setting": "contextRoot",
                 "value":  testContextRoot,
                 "socketEvent": eventConfigs.events.settingsChanged,
                 "eventKeys": ["operationId", "projectID", "contextRoot", "status"],
-                "afterHook": [afterHookContextRootTest]
+                "result": {
+                    "projectID": projData.projectID,
+                    "status": "success",
+                    "contextRoot": testContextRoot
+                },
+                "afterHook": [{
+                    "title": "Context Root After Hook: Reset project context root endpoint",
+                    "function": afterHookContextRootTest
+                }]
             },
             "combo4": {
                 "setting": "healthCheck",
                 "value":  testHealthCheck,
                 "socketEvent": eventConfigs.events.settingsChanged,
                 "eventKeys": ["operationId", "projectID", "name", "healthCheck", "status"],
-                "afterHook": [afterHookHealthCheckTest]
+                "result": {
+                    "projectID": projData.projectID,
+                    "status": "success",
+                    "healthCheck": testHealthCheck
+                },
+                "afterHook": [{
+                    "title": "Health Check After Hook: Reset project health check endpoint",
+                    "function": afterHookHealthCheckTest
+                }]
             },
             "combo5": {
                 "setting": "mavenProfiles",
                 "socketEvent": eventConfigs.events.settingsChanged,
-                "eventKeys": ["operationId", "projectID", "mavenProfiles", "status"]
+                "eventKeys": ["operationId", "projectID", "mavenProfiles", "status"],
+                "result": {
+                    "projectID": projData.projectID,
+                },
             },
             "combo6": {
                 "setting": "mavenProperties",
                 "socketEvent": eventConfigs.events.settingsChanged,
-                "eventKeys": ["operationId", "projectID", "mavenProperties", "status"]
+                "eventKeys": ["operationId", "projectID", "mavenProperties", "status"],
+                "result": {
+                    "projectID": projData.projectID,
+                },
             },
             "combo7": {
                 "setting": "ignoredPaths",
```

Comment:
are we testing the `ignoredPaths`, `contextRoot`, `healthCheck` settings? Not sure if I am seeing it in the PR view here?

Reply:
Yes, they are being tested. The if conditions in the test functions are there for specific settings because there are additional configurations for those settings